src/NuGet.Core/NuGet.Packaging/NupkgMetadata/NupkgMetadataFileFormat.cs

```diff
@@ -53,11 +53,11 @@ public static NupkgMetadataFile Read(TextReader reader, ILogger log, string path
             }
             catch (Exception ex)
             {
-                log.LogWarning(string.Format(CultureInfo.CurrentCulture,
+                var message = string.Format(CultureInfo.CurrentCulture,
                     Strings.Error_LoadingHashFile,
-                    path, ex.Message));
+                    path, ex.Message);
 
-                throw;
+                throw new Exception(message);
```

Comment:
```suggestion
                throw new Exception(message, ex);
```

It's good practise to always retain the original exception as an inner exception.